spec.emu

```diff
@@ -390,6 +407,31 @@ contributors: Google, Ecma International
     </emu-note>
   </emu-clause>
 
+  <emu-clause id="sup-temporal-calendardateextendedyear" type="abstract operation">
+    <h1>
+      CalendarDateExtendedYear (
+        _calendar_: a String,
+        _date_: a Temporal.PlainDateTime, Temporal.PlainDate, or Temporal.PlainYearMonth,
+      ): an integer
+    </h1>
+    <dl class="header">
+      <dt>description</dt>
+      <dd>It performs implementation-defined processing to find the year for the date corresponding to _date_ in the context of the calendar represented by _calendar_ relative to a well-defined epoch year for that calendar.</dd>
+    </dl>
+    <p>
+        This definition supersedes the definition provided in "Temporal proposal", #sec-temporal-calendardateextendedyear.
+    </p>
+    <emu-alg>
+      1. Let _year_ be _date_.[[ISOYear]].
+      1. If IsCalendarSupportEra(_calendar_) is *false*, then
+        1. Return _year_.
+      1. Let _r_ be the row in <emu-xref href="#table-eras"></emu-xref> which _calendar_ is the value of the Calendar column as well as the Era column.
+      1. Let _epochYear_ be the value given in the *"Epoch Year"* column of _r_.
+      1. Let _extendedYear_ be _year_ - _epochYear_.
```

Comment:
I think this doesn't work for lunar calendars like Hijri where the years are a different length than the solar year.

Basically I think we need to spec that "arithmetical year 1 in calendar X is the year that starts in ISO year Y". For example, arithmetical year 1 for Hijri is the year that starts in ISO year 622.

And then separately we say that arithmetical year is an `integer` where adding 1 results in the following year.